### DISCUSSION TITLE
Ollama-webui Chat Context or History?

### DISCUSSION THREAD

**Discussion by MrMWith** [open-webui/open-webui#207] [Q&A]
Where does Ollama-Webui save the chat context history?

**Comment by tjbck** [comment]
Hi, I'm not sure if I get your question. Could you elaborate a bit?

**Reply by MrMWith** [reply]
Well, with Ollama from the command prompt, if you look in the .ollama folder you  will see a history file.  This appears to be saving all or part of the chat sessions. Using Ollama-webui, the history file doesn't seem to exist so I assume webui is managing that someplace?

**Reply by tjbck** [reply]
The chat logs are being saved on the browser (combination of localStorage and indexedDB), most likely will add an option to save them on the backend in the near future, Thanks.

**Reply by tbendien** [reply]
hi there, I'm looking for a way to download all chat history form all users as the sysadmin. Is that possible now somehow please? www.bionic-gpt.com saves all chat history on a postgres DB that runs in a container as part of their standard deployment. It gets exposed on port 8080 and has its own yaml conifig file and credentials. Would be super cool if Ollama can do something like that!!!

**Reply by tjbck** [reply]
@tbendien we already have that feature for individuals, go to settings > chats > export chats to download your personal chat logs. I'll see what we can do for downloading entire chat logs in the db.

**Reply by tbendien** [reply]
Thanks TIm! I am using Ollama Web UI in schools and businesses, so we need the sysadmin to be able to download all chat logs and prevent users from permanently  deleting their chat history. This is so we can run analytics on the chats and also for audits etc. I imagine this is possible on Ollama Web UI? Thank you for a great project, its awesome.
